azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/features/NetworkSecurityGroupApiLiveTest.java

```diff
@@ -0,0 +1,135 @@
+import static org.testng.AssertJUnit.assertNull;
+
+@Test(groups = "live", singleThreaded = true)
+public class NetworkSecurityGroupApiLiveTest extends BaseAzureComputeApiLiveTest {
+
+   private String resourcegroup;
+   private static String DEFAULT_NSG_NAME = "testNetworkSecurityGroup";
+
+   private NetworkSecurityGroup createGroup() {
+      NetworkSecurityRule rule = NetworkSecurityRule.create("denyallout", null, null,
+                     NetworkSecurityRuleProperties.create("deny all out",
+                                                 Protocol.Tcp,
+                                                 "*",
+                                                 "*",
+                                                 "*",
+                                                 "*",
+                                                 Access.Deny,
+                                                 4095,
+                                                 Direction.Outbound));
+      ArrayList<NetworkSecurityRule> ruleList = new ArrayList<NetworkSecurityRule>();
+      ruleList.add(rule);
+      NetworkSecurityGroup nsg = NetworkSecurityGroup.create("samplensg", "westus", null,
+                                                             NetworkSecurityGroupProperties.create(ruleList, null,
+                                                                                                   null, null, null,
+                                                                                                   null),
+                                                             null);
+      return nsg;
+   }
+
+   @BeforeClass
+   @Override
+   public void setup() {
+      super.setup();
+      resourcegroup = getResourceGroupName();
+   }
+
+   @Test(groups = "live")
+   public void deleteNetworkSecurityGroupDoesNotExist() {
+      final NetworkSecurityGroupApi nsgApi = api.getNetworkSecurityGroupApi(resourcegroup);
+      URI uri = nsgApi.delete(DEFAULT_NSG_NAME);
+      assertNull(uri);
+   }
+
+   @Test(groups = "live", dependsOnMethods = "deleteNetworkSecurityGroupDoesNotExist")
+   public void createNetworkSecurityGroup() {
+      final NetworkSecurityGroup nsg = createGroup();
+      assertNotNull(nsg);
+
+      final NetworkSecurityGroupApi nsgApi = api.getNetworkSecurityGroupApi(resourcegroup);
+      NetworkSecurityGroup result = nsgApi.createOrUpdate(DEFAULT_NSG_NAME,
+                                                  nsg.location(),
+                                                  nsg.tags(),
+                                                  nsg.properties());
+      assertNotNull(result);
+   }
+
+   @Test(groups = "live", dependsOnMethods = "createNetworkSecurityGroup")
+   public void listNetworkSecurityGroups() {
+      final NetworkSecurityGroupApi nsgApi = api.getNetworkSecurityGroupApi(resourcegroup);
+      List<NetworkSecurityGroup> result = nsgApi.list();
+
+      assertNotNull(result);
+      assertEquals(result.size(), 1);
```

Comment:
It would be great to do a more robust test and add a check that ensures the one you created is present in the list, instead of just checking the size. Once the live test number starts to grow (and perhaps other tests reuse the same groups, etc) this will help a lot when it comes to maintain them.